modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2553,13 +2589,40 @@ def concat(
             new_columns = joined_index
             # TODO: Can optimize by combining if all dtypes are materialized
             new_dtypes = None
+            # If we have already cached the length of each row in at least one
+            # of the row's partitions, we can build new_lengths for the new
+            # frame. Typically, if we know the length for any partition in a
+            # row, we know the length for the first partition in the row. So
+            # just check the lengths of the first column of partitions.
+            new_lengths = []
+            if new_partitions.size > 0:
+                for part in new_partitions.T[0]:
+                    if part._length_cache is not None:
+                        new_lengths.append(part.length())
+                    else:
+                        new_lengths = None
+                        break
```

Comment:
I need the `if new_partitions.size > 0` in case `new_partitions` is empty. Without the check, I get `index 0 is out of bounds for axis 0 with size 0`. That happens in `modin/pandas/test/test_concat.py::test_concat_with_empty_frame`.

Reply:
Yes, I see.